### ISSUE TITLE
Unexpected prediction from SVM

### ISSUE THREAD

**Issue by aymanosman** [elixir-nx/scholar#312]
I get unexpected predictions from the SVM module.

This example is taken from the docs:

```elixir
x = Nx.tensor([[1.0, 2.0], [3.0, 2.0], [4.0, 7.0]])
y = Nx.tensor([1, 0, 1])
model = Scholar.Linear.SVM.fit(x, y, num_classes: 2)
Scholar.Linear.SVM.predict(model, x)
```

```
#Nx.Tensor<
  s32[3]
  [1, 1, 1]
>
```

I expected to see `[1, 0, 1]`.

Here is a livebook demonstrating the behaviour https://gist.github.com/aymanosman/aa20b3832148260d257cf43e2d497cc4.

**Comment by aymanosman** [comment]
> Why do you believe this behaviour is expected? Looking at Python, I get the same behaviour:

Because I am a n00b.

**Comment by aymanosman** [comment]
Although...

```python
from sklearn.linear_model import SGDClassifier
x = [[1.0, 2.0], [3.0, 2.0], [4.0, 7.0]]
y = [1, 0, 1]
clf = SGDClassifier()
clf.fit(x, y)
print(clf.predict(x))
# => [1 0 1]
```

**Comment by msluszniak** [comment]
Did you test other losses like hinge loss with a bigger value of the `C` parameter?

**Comment by aymanosman** [comment]
> Did you test other losses like hinge loss with a bigger value of the `C` parameter?

If you mean the python code, then hinge loss is the default.

I've tried a bunch of different parameters for `SGDClassifier`, some work and some don't but I have no idea what I'm doing so I'll close this issue for now.
